client/components/signup-site-preview/iframe-utils.ts

```diff
@@ -0,0 +1,123 @@
+// This module is for dealing with DOM elements from inside the the site preview
+// iframe. This means that not every modern JS feature will be available to use
+// because they won't have been polyfilled e.g. a NodeList from the iframe isn't
+// iterable using a for-of loop.
+/* eslint no-for-of-loops/no-for-of-loops: 2 */
+
+/**
+ * External dependencies
+ */
+import { forEach } from 'lodash';
+import { RefObject } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { isIE, getIframeSource, revokeObjectURL } from './utils';
+
+type IFrameRef = RefObject< HTMLIFrameElement >;
+
+/**
+ * @param iframe frame in which to set the content
+ * @param html html content to set in iframe
+ * @returns true if innerHTML was set successfully
+ */
+export function setEntryContentInnerHTML( iframe: IFrameRef, html: string ): boolean {
+	if ( ! iframe.current || ! iframe.current.contentWindow ) {
+		return false;
+	}
+	const element = iframe.current.contentWindow.document.querySelector( '.entry-content' );
+	if ( ! element ) {
+		return false;
+	}
+
+	element.innerHTML = html;
+	return true;
+}
+
+export function setTextContentAll( iframe: IFrameRef, selector: string, textContent: string ) {
```

Comment:
I was inspired by `querySelectorAll`, was trying to imply that it would set the text content for *all* the elements that matched the selector.

Sticking with `setIframeElementContent` seems better to me.